Source/Events.Store/CommittedAggregateEventsExtensions.cs

```diff
@@ -29,10 +29,16 @@ public static Contracts.CommittedAggregateEvents ToProtobuf(this CommittedAggreg
         protobuf.Events.AddRange(committedAggregateEvents.Select(_ => _.ToProtobuf()));
         return protobuf;
     }
-
+    
+    // TODO: THIS NEEDS SPECS
+    /// <summary>
+    /// Converts the <see cref="Contracts.CommittedAggregateEvents"/> to <see cref="CommittedAggregateEvents"/>.
+    /// </summary>
+    /// <param name="committedAggregateEvents">The committed events.</param>
+    /// <returns>The converted <see cref="CommittedAggregateEvents"/>.</returns>
     public static CommittedAggregateEvents ToCommittedEvents(this Contracts.CommittedAggregateEvents committedAggregateEvents)
     {
-        var version = committedAggregateEvents.AggregateRootVersion - (ulong)committedAggregateEvents.Events.Count;
+        var version = committedAggregateEvents.AggregateRootVersion + 1 - (ulong)committedAggregateEvents.Events.Count;
         return new CommittedAggregateEvents(
```

Comment:
I see this method is only used one place currently (in the Embeddings), but does it make sense? I'll need to make some specs for this :) @jakhog

Reply:
I'm not sure why we need to make the +1 change? And shouldn't it be possible to reuse the conversion that is used in FetchForAggregate? Should be the same thing right?

Reply:
Maybe I'm just stupid here. Let me have a look :)

Reply:
Naaah cannot use it since the stored event(in FetchForAggregate) has the aggregate root version on it. The contract version of committed aggregate event does not. I'm pretty sure the the +1 is correct here (the specs says so :P) I will make specs for it though

Reply:
I've written specs that should cover this method